services/files.js

```diff
@@ -1,17 +1,29 @@
 const assert = require('assert');
 
-const parse = require('./util/parseWikiUrl');
-
-function formatImagesInfoResponse(response) {
-  const { pages } = response;
-  assert.ok(!!pages, 'Wikimedia: No "url" option provided');
-
-  // TODO: use response wrapper to extract data (ImageInfo)
-  return Object.values(pages).map(page => {
-    const { title, imageinfo } = page;
-    const { url } = imageinfo[0];
-    return { title, url };
-  });
+const parseWikiUrl = require('./util/parseWikiUrl');
+
+async function getImageTitles({ client, title, wikiUrl }) {
+  const { pages } = await client.getResultsFromApi(title, 'images', wikiUrl);
+  assert.ok(pages, 'empty-response');
+  const { images = [] } = Object.values(pages)[0];
```

Comment:
Again, could it make sense to document that you're expecting exactly 1 page?

```javascript
assert.ok(pages.length === 1);
// or, a softer approach:
if (pages.length !== 1) console.warn(`expected exactly 1 page from API, instead received ${pages.length}`);
```

Reply:
I'm not sure how to test that 🙈 
`length` does not work since pages is an object, not an array... I could do `Object.values` as an intermediate step and then check that this contains only a single object 🤔

Reply:
See a previous comment at https://github.com/wmde/attribution-generator-api/pull/13/files#r247439689

Reply:
What type of error could we return to users in this case @juffel 🤔

Reply:
A great, thanx @pmeinhardt 🙏 

Solved it in a more 🚜 way 😹

Reply:
Force pushed and updated it to your version - prefer that over using `let`...